src/commands/image/imageSource/buildImageInAzure/TarFileStep.ts

```diff
@@ -16,7 +16,7 @@ export class TarFileStep extends AzureWizardExecuteStep<BuildImageInAzureImageSo
     public async execute(context: BuildImageInAzureImageSourceContext): Promise<void> {
         const id: number = Math.floor(Math.random() * Math.pow(10, idPrecision));
         const archive = `sourceArchive${id}.tar.gz`;
-        context.tarFilePath = Utils.joinPath(URI.parse(os.tmpdir()), archive).path;
+        context.tarFilePath = Utils.joinPath(URI.parse(os.tmpdir()), archive).fsPath;
```

Comment:
I'd do all the `fsPath` changes in another PR for the following reasons:

1. Makes the PR less girthy
2. If we need to revert one of these changes for some reason (either deploy API or fsPath), it'll be cumbersome to revert one thing without reverting everything. It doesn't happen often, but when we do have to revert things like that, I'm always grateful when they are separate commits/PRs.

Reply:
Yeah sure, I can pull some of these changes out, but I just wanted to test these changes together to ensure the two groups of changes played nicely together :)